visualization/src/point_cloud_handlers.cpp

```diff
@@ -686,56 +686,55 @@ pcl::visualization::PointCloudGeometryHandler<pcl::PCLPointCloud2>::getGeometry
 
   vtkSmartPointer<vtkFloatArray> data = vtkSmartPointer<vtkFloatArray>::New ();
   data->SetNumberOfComponents (3);
+  
   vtkIdType nr_points = cloud_->width * cloud_->height;
+  
+  if (!data->Resize(nr_points))
+  {
+    PCL_ERROR("[point_cloud_handlers::getGeometry] Failed to allocate space for points in VTK array.");
+    throw std::bad_alloc();
+  }
+    
 
   // Add all points
-  float dim;
-  vtkIdType j = 0;    // true point index
-  std::vector<float> pts (nr_points * 3);
   int point_offset = 0;
 
+  const float* ptr = nullptr;
```

Comment:
Move this into the for loops please

Reply:
Then I have to add two? Each in one for loop.

Reply:
The first invocation can have an extra `const float*`. No need to declare it separately.